airbyte-server/src/main/java/io/airbyte/server/RequestLogger.java

```diff
@@ -0,0 +1,141 @@
+import javax.ws.rs.core.Context;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.glassfish.jersey.message.MessageUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+// todo: make sure handler code has MDC functional
+
+public class RequestLogger implements ContainerRequestFilter, ContainerResponseFilter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RequestLogger.class);
+
+  @Context
+  private HttpServletRequest servletRequest;
+
+  private String requestBody = null;
+
+  private final Map<String, String> mdc;
+
+  public RequestLogger(Map<String, String> mdc) {
+    this.mdc = mdc;
+  }
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    if (requestContext.getMethod().equals("POST")) {
+      // hack to refill the entity stream so it doesn't interfere with other operations
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      IOUtils.copy(requestContext.getEntityStream(), baos);
+      InputStream entity = new ByteArrayInputStream(baos.toByteArray());
+      requestContext.setEntityStream(new ByteArrayInputStream(baos.toByteArray()));
+      // end hack
+
+      requestBody = IOUtils.toString(entity, MessageUtils.getCharset(requestContext.getMediaType()));
+    }
+  }
+
+  @Override
+  public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+    MDC.setContextMap(mdc);
+
+    String remoteAddr = servletRequest.getRemoteAddr();
+    String method = servletRequest.getMethod();
+    String url = servletRequest.getRequestURI();
+    int status = responseContext.getStatus();
+
+    StringBuilder logBuilder = new StringBuilder()
+        .append("REQ ")
+        .append(remoteAddr)
+        .append(" ")
+        .append(method)
+        .append(" ")
+        .append(status)
+        .append(" ")
+        .append(url);
+
+    if (method.equals("POST") && requestBody != null && !requestBody.equals("")) {
+      logBuilder
+          .append(" - ")
+          .append(redactSensitiveInfo(requestBody));
+    }
+
+    if (HttpStatus.isClientError(status) || HttpStatus.isServerError(status)) {
+      LOGGER.error(logBuilder.toString());
+    } else {
+      LOGGER.info(logBuilder.toString());
+    }
+  }
+
+  private static final Set<Class<?>> SENSITIVE_INFO_REQUESTS = Set.of(
+      SourceCreate.class,
+      SourceUpdate.class,
+      SourceRecreate.class,
+      DestinationCreate.class,
+      DestinationUpdate.class,
+      DestinationRecreate.class);
+
+  private static String redactSensitiveInfo(String requestBody) {
+    Optional<JsonNode> jsonNode = Jsons.tryDeserialize(requestBody);
+
+    if (jsonNode.isPresent()) {
+      for (Class<?> sensitiveClass : SENSITIVE_INFO_REQUESTS) {
+        Optional<?> object = Jsons.tryObject(jsonNode.get(), sensitiveClass);
+        if (object.isPresent()) {
```

Comment:
is it always the `configuration` field that contains the sensitive info? if so i might argue that it would be nice to print everything but that field so we can have a sanity check that the rest looks good. but if it's not dead simple to do, ignore.

Reply:
Yeah it looks like `connectionConfiguration` only right now. I'm going to remove it, but only from the top level for now.